Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,45 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python2?
+        requirements.any? do |req|
+          req.is_a?(PythonRequirement) && !req.is_a?(Python3Requirement) &&
+            (req.required? || build.with?(req))
+        end ||
+          deps.any? do |dep|
+            dep.name == "python" && (dep.required? || build.with?(dep))
+          end
+      end
```

Comment:
I agree that it does a decent job providing a better default and I'm happy with this. Just wanted to point out that it won't be able to get all cases right automagically.